### PR TITLE
os/bluestore/NVMEDevice: change write_bl to bl

### PR DESCRIPTION
**Pull request by optimistyzy**
With this patch, both read and write can use this field in
task.

Signed-off-by: Ziye Yang <optimistyzy@gmail.com>
Signed-off-by: Pan Liu <wanjun.lp@alibaba-inc.com>

### CODE REVIEW

**Comment by optimistyzy** [comment]
@liupan1111

**Comment by optimistyzy** [comment]
Jenkins, retest this please

**Comment by optimistyzy** [comment]
@yuyuyu101  @liupan1111   This is a cleanup.

File: src/os/bluestore/NVMEDevice.cc [hunk 1]
```diff
@@ -323,6 +323,11 @@ struct Task {
     io_request.nseg = 0;
   }
 
+  void copy_to_buf(bufferptr p, uint64_t off, uint64_t len) {
```
Comment (yuyuyu101) [inline src/os/bluestore/NVMEDevice.cc]: bufferptr &&p

File: src/os/bluestore/NVMEDevice.cc [hunk 2]
```diff
@@ -1090,11 +1095,10 @@ int NVMEDevice::aio_read(
   Task *t = new Task(this, IOCommand::READ_COMMAND, off, len);
 
   bufferptr p = buffer::create_page_aligned(len);
-  pbl->append(p);
+  pbl->append(t->bl);
```
Comment (yuyuyu101) [inline src/os/bluestore/NVMEDevice.cc]: std::move(

File: src/os/bluestore/NVMEDevice.cc [hunk 3]
```diff
@@ -1056,9 +1061,9 @@ int NVMEDevice::read(uint64_t off, uint64_t len, bufferlist *pbl,
   bufferptr p = buffer::create_page_aligned(len);
   int r = 0;
   t->ctx = ioc;
-  char *buf = p.c_str();
-  t->fill_cb = [buf, t]() {
-    t->copy_to_buf(buf, 0, t->len);
+  pbl->append(t->bl);
```
Comment (yuyuyu101) [inline src/os/bluestore/NVMEDevice.cc]: std::move(

File: src/os/bluestore/NVMEDevice.cc (hunk 2 above)
Comment (optimistyzy) [inline src/os/bluestore/NVMEDevice.cc]: Could you explain why use std:move() sice at this time, there is no data copy yet. Also in kernelDevice.cc, it also directly use append.

File: src/os/bluestore/NVMEDevice.cc (hunk 1 above)
Comment (optimistyzy) [inline src/os/bluestore/NVMEDevice.cc]: why &&p, I think that bufferptr p is sufficient

Comment (yuyuyu101) [inline src/os/bluestore/NVMEDevice.cc]: I don't think so... @tchaikov plz correct me if wrong

**Comment by optimistyzy** [comment]
@yuyuyu101 @liupan1111   Could you review this patch again?

**Pull request merged**